PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ["ir.attachment"]
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users']

Total Changes: 9

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/web_unsplash/models/ir_attachment.py
+++ b/addons/web_unsplash/models/ir_attachment.py
@@ -5,8 +5,7 @@ from odoo import models
 
 
 class IrAttachment(models.Model):
-
-    _inherit = ["ir.attachment"]
+    _inherit = "ir.attachment"
 
     def _can_bypass_rights_on_media_dialog(self, **attachment_data):
         # We need to allow and sudo the case of an "url + file" attachment,

--- a/addons/web_unsplash/models/ir_qweb_fields.py
+++ b/addons/web_unsplash/models/ir_qweb_fields.py
@@ -4,7 +4,7 @@ from odoo import models, api
 
 
 class IrQwebFieldImage(models.AbstractModel):
-    _inherit = ['ir.qweb.field.image']
+    _inherit = 'ir.qweb.field.image'
 
     @api.model
     def from_html(self, model, field, element):

--- a/addons/web_unsplash/models/res_config_settings.py
+++ b/addons/web_unsplash/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     unsplash_access_key = fields.Char("Access Key", config_parameter='unsplash.access_key')
     unsplash_app_id = fields.Char("Application ID", config_parameter='unsplash.app_id')

--- a/addons/web_unsplash/models/res_users.py
+++ b/addons/web_unsplash/models/res_users.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     def _can_manage_unsplash_settings(self):
         self.ensure_one()
